lib/node/rpc.js

```diff
@@ -2626,6 +2610,16 @@ class RPC extends RPCBase {
 
     let attempt = this.attempt;
 
+    // This check is here to make sure a miner does not
+    // accidentally OOM the node. This means the merkle
+    // map is good for 2 hours without a block.
+    if (this.merkleMap.size >= 720) {
```

Comment:
Just to be clear, this is because we won't create a new template / add to `merkleMap` until at least 10 seconds pass. So if there are 720 items in the map, that means 7,200 seconds have passed (2 hours).

```js
      if (util.now() - this.lastActivity > 10)
        this.refreshBlock();
```